go/integTest/batch_test.go

```diff
@@ -122,17 +124,273 @@ func CreateStringTest(batch *pipeline.ClusterBatch, isAtomic bool) BatchTestData
+
+	if serverVer >= "7.0.0" {
+		batch.SortReadOnly(slotHashedKey1)
+		testData = append(
+			testData,
+			CommandTestData{ExpectedResponse: []any{"1", "2", "3"}, TestName: "SortReadOnly(slotHashedKey1)"},
+		)
+	}
+
+	batch.Del([]string{slotHashedKey1})
+	testData = append(testData, CommandTestData{ExpectedResponse: int64(1), TestName: "Del(slotHashedKey1)"})
+	batch.LPush(slotHashedKey1, []string{"c", "b", "a"})
+	testData = append(testData, CommandTestData{ExpectedResponse: int64(3), TestName: "LPush(slotHashedKey1, [c, b, a])"})
+
+	if serverVer >= "7.0.0" {
+		batch.SortReadOnlyWithOptions(slotHashedKey1, *options.NewSortOptions().SetIsAlpha(true))
+		testData = append(
+			testData,
+			CommandTestData{
+				ExpectedResponse: []any{"a", "b", "c"},
+				TestName:         "SortReadOnlyWithOptions(slotHashedKey1, {Alpha: true})",
+			},
+		)
+	}
+
+	batch.Wait(0, 0)
+	testData = append(testData, CommandTestData{ExpectedResponse: int64(0), CheckType: true, TestName: "Wait(0, 0)"})
+
+	batch.Del([]string{slotHashedKey1, slotHashedKey2})
+	testData = append(testData, CommandTestData{ExpectedResponse: int64(2), TestName: "Del(slotHashedKey1, slotHashedKey2)"})
+	batch.Set(slotHashedKey1, "value")
+	testData = append(testData, CommandTestData{ExpectedResponse: "OK", TestName: "Set(slotHashedKey1, value)"})
+	batch.Copy(slotHashedKey1, slotHashedKey2)
+	testData = append(testData, CommandTestData{ExpectedResponse: true, TestName: "Copy(slotHashedKey1, slotHashedKey2)"})
+	batch.Get(slotHashedKey2)
+	testData = append(testData, CommandTestData{ExpectedResponse: "value", TestName: "Get(slotHashedKey2) after Copy"})
+
+	batch.Del([]string{slotHashedKey1, slotHashedKey2})
+	testData = append(testData, CommandTestData{ExpectedResponse: int64(2), TestName: "Del(slotHashedKey1, slotHashedKey2)"})
+	batch.Set(slotHashedKey1, "value1")
+	testData = append(testData, CommandTestData{ExpectedResponse: "OK", TestName: "Set(slotHashedKey1, value1)"})
+	batch.Set(slotHashedKey2, "value2")
+	testData = append(testData, CommandTestData{ExpectedResponse: "OK", TestName: "Set(key2, value2)"})
+	batch.CopyWithOptions(slotHashedKey1, slotHashedKey2, *options.NewCopyOptions().SetReplace())
+	testData = append(
+		testData,
+		CommandTestData{
+			ExpectedResponse: true,
+			TestName:         "CopyWithOptions(slotHashedKey1, slotHashedKey2, ReplaceDestination)",
+		},
+	)
+	batch.Get(slotHashedKey2)
+	testData = append(
+		testData,
+		CommandTestData{ExpectedResponse: "value1", TestName: "Get(slotHashedKey2) after CopyWithOptions"},
+	)
+
+	return BatchTestData{CommandTestData: testData, TestName: "Generic Base commands"}
+}
+
+type BatchTestDataProvider func(*pipeline.ClusterBatch, bool, string) BatchTestData
 
 func GetCommandGroupTestProviders() []BatchTestDataProvider {
 	return []BatchTestDataProvider{
 		CreateStringTest,
 		// more command groups here
+		CreateGenericBaseTests,
 	}
 }
 
 type CommandTestData struct {
 	ExpectedResponse any
+	CheckType        bool
```

Comment:
```suggestion
	CheckTypeOnly    bool
```
Please rename as we discussed